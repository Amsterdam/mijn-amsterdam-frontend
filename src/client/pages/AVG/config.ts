import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

export const displayPropsAanvragen: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
};

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
];
