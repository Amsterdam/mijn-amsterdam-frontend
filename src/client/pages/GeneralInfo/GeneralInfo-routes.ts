import { GeneralInfo } from './GeneralInfo';

export const GeneralInfoRoute = { route: '/uitleg', Component: GeneralInfo };
export const GeneralInfoRoutes = [GeneralInfoRoute];

export const GENERAL_INFO_PAGE_DOCUMENT_TITLE = 'Dit ziet u in Mijn Amsterdam';
