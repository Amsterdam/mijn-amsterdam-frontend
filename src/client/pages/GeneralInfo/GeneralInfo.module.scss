@import '../../styles/_global.scss';

.GeneralInfo {

}
