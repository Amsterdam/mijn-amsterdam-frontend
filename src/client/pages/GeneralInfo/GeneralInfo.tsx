import React from 'react';
import { isError, isLoading } from '../../../universal/helpers';
import {
  Alert,
  InnerHtml,
  LoadingContent,
  PageContent,
  PageHeading,
  TextPage,
} from '../../components';
import { useAppStateGetter } from '../../hooks/useAppState';

export default function GeneralInfo() {
  const { CMS_CONTENT } = useAppStateGetter();
  const generalInfo = CMS_CONTENT.content?.generalInfo;

  return (
    <TextPage>
      <PageHeading isLoading={isLoading(CMS_CONTENT)}>
        {generalInfo?.title || 'Over Mijn Amsterdam'}
      </PageHeading>
      <PageContent>
        {isError(CMS_CONTENT) ||
          (generalInfo === null && (
            <Alert type="warning">
              <p>We kunnen de inhoud van deze pagina nu niet weergeven.</p>
            </Alert>
          ))}
        {isLoading(CMS_CONTENT) && <LoadingContent />}
        {generalInfo?.content && <InnerHtml>{generalInfo?.content}</InnerHtml>}
      </PageContent>
    </TextPage>
  );
}
