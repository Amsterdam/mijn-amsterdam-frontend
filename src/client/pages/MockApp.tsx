import { RecoilRoot, RecoilRootProps } from 'recoil';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

interface MockAppProps {
  routePath: string;
  routeEntry: string;
  initializeState?: RecoilRootProps['initializeState'];
  component: React.Component | any;
}

export default function MockApp({
  routePath,
  routeEntry,
  initializeState,
  component,
}: MockAppProps) {
  return (
    <RecoilRoot initializeState={initializeState}>
      <MemoryRouter initialEntries={[routeEntry]}>
        <Route path={routePath} component={component} />
      </MemoryRouter>
    </RecoilRoot>
  );
}
