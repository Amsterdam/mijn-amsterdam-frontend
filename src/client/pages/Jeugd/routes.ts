import { JeugdDetail } from './JeugdDetail';
import { JeugdThemaPagina } from './JeugdThema';
import { FeatureToggle } from '../../../universal/config/feature-toggles';

const baseRoute = '/jeugd';

export const JeugdRoutes = [
  {
    route: baseRoute,
    Component: JeugdThemaPagina,
    isActive: FeatureToggle.zorgnedLeerlingenvervoerActive,
  },
  {
    route: `${baseRoute}/voorziening/:id`,
    Component: JeugdDetail,
    isActive: FeatureToggle.zorgnedLeerlingenvervoerActive,
  },
];
