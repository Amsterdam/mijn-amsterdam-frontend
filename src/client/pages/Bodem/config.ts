import { generatePath } from 'react-router-dom';

import { LoodMetingFrontend } from '../../../server/services/bodem/types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';

const displayPropsAanvragen = {
  detailLinkComponent: 'Adres',
  datumAanvraagFormatted: 'Aangevraagd',
  status: 'Status',
};

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    ...tableConfigBase,
    title: 'Lopende aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => !bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
  },
  [listPageParamKind.completed]: {
    ...tableConfigBase,
    title: 'Afgehandelde aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.completed,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    title: 'Meer informatie over lood in de bodem.',
    to: 'https://www.amsterdam.nl/wonen-leefomgeving/bodem/lood-grond/',
  },
] as const;
