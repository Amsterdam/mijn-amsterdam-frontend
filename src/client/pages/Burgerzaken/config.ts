import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { IdentiteitsbewijsFrontend, LinkProps } from '../../../universal/types';
import { withOmitDisplayPropsForSmallScreens } from '../../components/Table/helpers';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

const displayPropsBase: DisplayProps<
  WithDetailLinkComponent<IdentiteitsbewijsFrontend>
> = {
  detailLinkComponent: 'Type',
  datumAfloopFormatted: 'Geldig tot',
};

const listPageParamKind = {
  identiteitsbewijzen: 'identiteitsbewijzen',
};

const displayProps = withOmitDisplayPropsForSmallScreens(displayPropsBase, [
  'datumAfloopFormatted',
]);

export const tableConfig = {
  [listPageParamKind.identiteitsbewijzen]: {
    title: 'Mijn reisdocumenten',
    displayProps,
    sort: dateSort('datumAfloop', 'desc'),
    listPageRoute: generatePath(AppRoutes['BURGERZAKEN/LIST'], {
      kind: listPageParamKind.identiteitsbewijzen,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/burgerzaken',
    title: 'Overzicht en aanvragen bij burgerzaken',
  },
] as const;

export const backLinkDetailPage = AppRoutes.BURGERZAKEN;
