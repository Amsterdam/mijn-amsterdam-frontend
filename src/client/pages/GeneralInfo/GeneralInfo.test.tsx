import { render } from '@testing-library/react';
import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';

import { AppRoutes } from '../../../universal/config/routes';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import GeneralInfo from './GeneralInfo';

const testState: any = {
  CMS_CONTENT: {
    status: 'OK',
    content: {
      generalInfo: {
        content: '<p>Dingen! <a href="http://example.org">Linkje</a></p>',
        title: 'Algemene info',
      },
    },
  },
};

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

describe('<GeneralInfo />', () => {
  const routeEntry = generatePath(AppRoutes.GENERAL_INFO);
  const routePath = AppRoutes.GENERAL_INFO;

  const Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={GeneralInfo}
      initializeState={initializeState}
    />
  );

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
