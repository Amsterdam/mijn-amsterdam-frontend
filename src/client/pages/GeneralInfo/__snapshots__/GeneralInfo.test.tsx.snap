// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GeneralInfo /> > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/uitleg"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  Algemene info
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h1
          class="ams-heading ams-heading--1 _PageHeading_c5a3f1"
        >
          Algemene info
        </h1>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <p
        class="ams-paragraph ams-mb-m"
      >
        Dingen! 
        <a
          class="ams-link"
          href="http://example.org"
        >
          Linkje
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
